src/odemis/odemisd/mdupdater.py

```diff
@@ -213,32 +213,32 @@ def updatePixelDensity(unused, lens=lens, comp=comp, binva=binva):
 
         # update pole position (if available), taking into account the binning
         if model.hasVA(lens, "polePosition"):
-            def updatePolePos(unused, lens=lens, comp=comp):
```

Comment:
I think you can drop `lens=lens`, indeed, but you might want to double or triple check. To understand the potential issue, have a look at https://stackoverflow.com/questions/12423614/local-variables-in-python-nested-functions .
(And the behaviour changed in Python 3, to help).

I *think* that you can drop it here, because as `lens` is defined as argument of the method, and never changed in the method. The reason that I had put it like this is that it's so hard to know when it's needed and when it's not, and that it depends on the code around the function, for "safety", I have the habit to always redefine the scope of the variables used in a local function. BTW, if you can drop `lens=lens`, you should be able to do the same with `comp_affected`. 

Still, I'd suggest to leave all the `lens=lens` as-is. Maybe add a comment explaining why that is, on the first line of the file with this "trick".